app/models/manageiq/providers/autosde/inventory/parser/storage_manager.rb

```diff
@@ -43,6 +45,54 @@ def storage_resources
     end
   end
 
+  def san_addresses
+    port_type = {
+      "ISCSI"  => "IscsiAddress",
+      "FC"     => "FiberChannelAddress",
+      "NVMeFC" => "NvmeAddress"
+    }
+
+    san_addresses = collector.storage_hosts.flat_map do |host_initiator|
+      host_initiator.addresses.flat_map do |address|
+        {
+          :ems_ref              => address.uuid,
+          :host_initiators_uuid => host_initiator.uuid,
+          :type                 => port_type[address.port_type],
+          :iqn                  => address.iqn,
+          :chap_name            => address.chap_name,
+          :chap_secret          => address.chap_secret,
+          :wwpn                 => address.wwpn
+        }
+      end
+    end
+
+    san_addresses.each do |san_address_hash|
+      host_initiators_uuid = san_address_hash.delete(:host_initiators_uuid)
+      persister.san_addresses.build(
+        **san_address_hash,
+        :owner => persister.host_initiators.lazy_find(host_initiators_uuid)
+      )
+    end
+  end
+
+  def host_initiators
+    host_initiators = collector.storage_hosts.map do |host_initiator|
```

Comment:
Same thing here, this can just become

```ruby
collector.storage_hosts.each do |host_initiator|
  persister.host_initiators.build(
    :name             => host_initiator.name,
    :ems_ref          => host_initiator.uuid,
    :physical_storage => persister.physical_storages.lazy_find(host_initiator.storage_system)
  )
end
```

Reply:
great. done!